### PR TITLE
Fix bad_grouping_or_ordering_0 test

### PR DESCRIPTION
**Pull request by sinkingpoint**
This test is really janky in that it actually tests 4 things,
rather than just the metric interweaving it proports to test:

- "A Histogram MetricPoint MUST contain at least one bucket (the given one doesn't)
- "Histogram MetricPoints MUST have at least a bucket with an +Inf threshold" (the given one doesn't)
- The sum without a count thing I mentioned in #229 (the histograms have sums without counts)
- The metric interweaving (what it's meant to be testing)

This commit fixes the first three to make this only test one thing

### CODE REVIEW

**Comment by brian-brazil** [comment]
That would make sense, the test was designed to test the specific code of the Python parser. Can you fix the Go validator so it correctly rejects this?

**Comment by brian-brazil** [comment]
I think you forgot to push the changes to the Go code itself.

**Comment by sinkingpoint** [comment]
Sure. Think that commit should do it. It's not the cleanest, but it seems to work

File: src/validator/validator.go [hunk 1]
```diff
@@ -535,6 +552,24 @@ func (v *OpenMetricsValidator) compareMetricFamilies(mfn string, last, cur *metr
...
+func getIgnoredLabels(name string, cur *metricFamily) []string {
+	ignored := []string{}
+	ignored = append(ignored, "__name__")
+	switch cur.MetricType() {
+	case textparse.MetricTypeHistogram:
+	case textparse.MetricTypeGaugeHistogram:
+		if strings.HasSuffix(name, "_bucket") {
+			ignored = append(ignored, labels.BucketLabel)
+		}
+	case textparse.MetricTypeStateset:
```
Comment (brian-brazil) [inline src/validator/validator.go:564]: I think you meant summary. Also there's no suffix on a quantile, you want to check the sample metric name matches the metric family name.

File: src/validator/validator.go [hunk 2]
```diff
@@ -535,6 +552,24 @@ func (v *OpenMetricsValidator) compareMetricFamilies(mfn string, last, cur *metr
 	}
 }
 
+func getIgnoredLabels(name string, cur *metricFamily) []string {
+	ignored := []string{}
+	ignored = append(ignored, "__name__")
+	switch cur.MetricType() {
+	case textparse.MetricTypeHistogram:
```
Comment (brian-brazil) [inline src/validator/validator.go:559]: Fallthroughs must be explicit in Go

**Comment by sinkingpoint** [comment]
Fixed those. Thanks for all your guidance @brian-brazil

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**